File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/overload_extend2_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'overload_extend2'

f = Overload_extend2::Foo.new

raise RuntimeError if f.test(3) != 1
raise RuntimeError if f.test("hello") != 2
raise RuntimeError if f.test(3.5,2.5) != 3

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/overload_extend2_runme.py
import overload_extend2

f = overload_extend2.Foo()
if f.test(3) != 1:
    raise RuntimeError
if f.test("hello") != 2:
    raise RuntimeError
if f.test(3.5, 2.5) != 3:
    raise RuntimeError
if f.test("hello", 20) != 1020:
    raise RuntimeError
if f.test("hello", 20, 100) != 120:
    raise RuntimeError

# C default args
if f.test(f) != 30:
    raise RuntimeError
if f.test(f, 100) != 120:
    raise RuntimeError
if f.test(f, 100, 200) != 300:
    raise RuntimeError
